### ISSUE TITLE
keyword arguments for `jnp.clip` have changed for Jax versions >= 0.10.0

### ISSUE THREAD

**Issue by CKrawczyk** [Herculens/herculens#95]
From the Jax change logs:

>The deprecated keyword arguments `a`, `a_min`, and `a_max` to `jax.numpy.clip` have been removed.

In current versions of Jax `a_min` is now `min` and `a_max` is now `max`.  These new keywords have been supported since Jax v0.4.37 so there would be no need to change the minimum version of Jax in the requirements.  More testing would need to be done to ensure everything else works with higher Jax versions (some of the new features in Jax v0.11 look like they might be useful for further optimizing some of the new code I have been working on so I will do some testing to see what is needed to make everything compatible).

It looks like the instances that need updating are:

- https://github.com/Herculens/herculens/blob/fd2e8754507ec213d66e7d355f3da162f4cdc1d0/herculens/Inference/loss.py#L45
- https://github.com/Herculens/herculens/blob/fd2e8754507ec213d66e7d355f3da162f4cdc1d0/herculens/MassModel/Profiles/point_mass.py#L39
- https://github.com/Herculens/herculens/blob/fd2e8754507ec213d66e7d355f3da162f4cdc1d0/herculens/MassModel/Profiles/point_mass.py#L54
- https://github.com/Herculens/herculens/blob/fd2e8754507ec213d66e7d355f3da162f4cdc1d0/herculens/Inference/sensitivity_mapping.py#L100
- https://github.com/Herculens/herculens/blob/fd2e8754507ec213d66e7d355f3da162f4cdc1d0/herculens/Inference/sensitivity_mapping.py#L166
- https://github.com/Herculens/herculens/blob/fd2e8754507ec213d66e7d355f3da162f4cdc1d0/herculens/MassModel/Profiles/nie.py#L249
- https://github.com/Herculens/herculens/blob/fd2e8754507ec213d66e7d355f3da162f4cdc1d0/herculens/Inference/legacy/parameters.py#L271
